scheduler.go

```diff
@@ -0,0 +1,98 @@
+/*
+ * File: scheduler.go
+ *
+ * Description: functions and definitions for the global scheduler
+ */
+
+package main
+
+// The element definition, which are individuals in the queue.
+type Element struct {
+	node *Node
+}
+
+// The scheduler definition, which holds the queue.
+type Scheduler struct {
+
+	// holds the current queue
+	Queue []*Element
+}
+
+//! Sync the global scheduler with the node manager.
+/*
+ * @param    string    primed node uuid
+ * @param    *Node[]   array of node pointers
+ *
+ * @return   none
+ */
+func syncScheduler(hid string, nodes []*Node) {
+
+	// Assign memory for the scheduler queue.
+	scheduler.Queue = make([]*Element, 0)
```

Comment:
Ah, the nature of languages that hide garbage collection from the developer. Much appreciated :)

https://github.com/rbisewski/lclusterd/commit/7b474ac8ba2f9df23ddcefb66ead37d05d12cbae